tests/test_io.py

```diff
@@ -17,6 +19,64 @@
 TEST_DIR = os.path.join(os.path.dirname(__file__), "test_files")
 
 
+class TestGetLineEnding:
```

Comment:
not sure if `_get_line_ending` is meant to handle that but i noticed there's no test for a file with mixed unix/windows line endings which e.g. can happen if you concatenate files from different OSes. one option would be for the function to return a tuple of all the line endings it finds but maybe that's overkill here

Reply:
> not sure if `_get_line_ending` is meant to handle that but i noticed there's no test for a file with mixed unix/windows line endings 

@hongyi-zhao https://github.com/materialsvirtuallab/monty/pull/712#issuecomment-2329133801 mentioned this possibility, but I don't think this should be handled by `get_line_ending`. I would expect a user to pass a correctly formatted file into this function (mixed line ending is certainly not a correct format), if not the use should fix the format before passing the file in.

> one option would be for the function to return a tuple of all the line endings it finds but maybe that's overkill here

I'm afraid that would be quite expensive to walk through the file again, perhaps still separate the functionalities, and leave format fixing to another function/tool.